src/seedu/addressbook/storage/AddressBookDecoder.java

```diff
@@ -0,0 +1,110 @@
+package seedu.addressbook.storage;
+
+import static seedu.addressbook.parser.Parser.PERSON_DATA_ARGS_FORMAT;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.storage.StorageFile.StorageOperationException;
+
+/**
+ * Decodes the storage data file into an AddressBook object.
+ */
+public class AddressBookDecoder {
+
+    /**
+     * Decodes the AddressBook in the storage data file.
+     *
+     * @param encodedAddressBook strings to be decoded.
+     * @return decoded AddressBook.
+     * @throws IllegalValueException if any field is invalid.
+     * @throws StorageOperationException if cannot decode any.
+     */
+    public static AddressBook decodeAddressBook(List<String> encodedAddressBook)
+            throws IllegalValueException, StorageOperationException {
+        final List<Person> successfullyDecodedPersons = decodePersonsFromStrings(encodedAddressBook);
+        return new AddressBook(new UniquePersonList(successfullyDecodedPersons));
+    }
+
+    /**
+     * Decodes persons from a list of string representations.
+     *
+     * @param encodedPersons strings to be decoded.
+     * @return List containing all decoded persons.
+     * @throws IllegalValueException if any field is invalid.
+     * @throws StorageOperationException if cannot decode any.
+     */
+    private static List<Person> decodePersonsFromStrings(List<String> encodedPersons)
+            throws IllegalValueException, StorageOperationException {
+        final List<Person> decodedPersons = new ArrayList<>();
+        for (String encodedPerson : encodedPersons) {
+            decodedPersons.add(decodePersonFromString(encodedPerson));
+        }
+        return decodedPersons;
+    }
+
+    /**
+     * Decodes a person from it's supposed string representation.
+     *
+     * @param encodedPerson string to be decoded.
+     * @return decoded person.
+     * @throws IllegalValueException if any field is invalid.
+     * @throws StorageOperationException if encodedPerson cannot be decoded.
+     */
+    private static Person decodePersonFromString(String encodedPerson)
+            throws IllegalValueException, StorageOperationException {
+        final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(encodedPerson);
+        // Validate encodedPerson string format
+        if (!matcher.matches()) {
```

Comment:
Since line 69 uses `PERSON_DATA_ARGS_FORMAT` and line 71 uses `!matcher.matches()`, the comment on Line 70 seems unnecessary because it's clear that line 69 & 71 are performing some kind of validation.